spec/section_spec.rb

```diff
@@ -44,24 +63,71 @@ class Page < SitePrism::Page; end
     expect(SitePrism::Section).to respond_to(:elements)
   end
 
-  it 'passes a given block to Capybara.within' do
-    expect(Capybara).to receive(:within).with('div')
+  describe 'instance' do
+    it 'passes a given block to Capybara.within' do
+      expect(Capybara).to receive(:within).with(locator)
 
-    SitePrism::Section.new(a_page, 'div') { 1 + 1 }
-  end
+      section_with_block
+    end
 
-  it 'does not require a block' do
-    expect(Capybara).not_to receive(:within)
+    it 'does not require a block' do
+      expect(Capybara).not_to receive(:within)
 
-    SitePrism::Section.new(a_page, 'div')
-  end
+      subject
+    end
 
-  describe 'instance' do
-    subject(:section) { SitePrism::Section.new('parent', 'locator') }
+    it 'evaluates visibility by delegating through root_element' do
+      expect(locator).to receive(:visible?)
 
-    it 'responds to javascript methods' do
-      expect(section).to respond_to(:execute_script)
-      expect(section).to respond_to(:evaluate_script)
+      subject.visible?
+    end
+
+    it 'obtains the text of a section by delegating through root_element' do
```

Comment:
Thank you for adding these tests!

Since you're testing separate public methods of Section here (the API) it might be clearer to name your tests to make that super obvious.  
e.g.

```
describe '#visible?' do
  it 'delegates through root_element'
```

If you feel that's too long, I won't object to combining the two lines into one for now.  I do think this makes it clearer that *this test* is the test for the `#visible` method.

Reply:
Yes, it would be a good idea to, in the near term, refactor specs so that the describe blocks _explicitly_ state which public API method they are testing. http://www.betterspecs.org/#describe